gui/pregnancy.lua

```diff
@@ -3,18 +3,32 @@ local widgets = require('gui.widgets')
 
 PregnancyGui = defclass(PregnancyGui, widgets.Window)
 PregnancyGui.ATTRS {
-    frame_title='My Window',
-    frame={w=50, h=45},
+    frame_title='Pregnancy manager',
+    frame={w=64, h=35},
     resizable=true, -- if resizing makes sense for your dialog
     resize_min={w=50, h=20}, -- try to allow users to shrink your windows
 }
 
 function PregnancyGui:init()
-    self.mother = false
+    if dfhack.gui.getSelectedUnit(true).sex == df.pronoun_type.she then
+        self.mother = dfhack.gui.getSelectedUnit(true)
+    else self.mother = false
+    end
     self.father = false
     self.father_historical = false
     self.msg = {}
-    -- self.success = false
+
+    local term_options = {}
+    local term_index = {}
+    local months
+    for months=0,10 do
+        -- table.insert(term_options,{label=('%s months'):format(months),value=months}) --I tried this to add labels, probably doing something wrong, it broke the range widget
```

Comment:
I notice below you were getting the label to index an array. this should work fine if you get the option value instead